### PR TITLE
HCE-792: Add Project ID to Consul Cluster Resource

### PR DESCRIPTION
**Pull request by paraspraj**
### :hammer_and_wrench: Description
 Add project_id to Consul Cluster resources.

### CODE REVIEW

File: internal/provider/data_source_consul_agent_helm_config.go [hunk 1]
```diff
@@ -83,13 +84,20 @@ func dataSourceConsulAgentHelmConfig() *schema.Resource {
...
-			// Optional
+			// Optional inputs
 			"expose_gossip_ports": {
 				Description: "Denotes that the gossip ports should be exposed.",
 				Type:        schema.TypeBool,
 				Optional:    true,
 				Default:     false,
 			},
+			"project_id": {
+				Description:  "The ID of the HCP project where the HCP Consul Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/data_source_consul_agent_helm_config.go:95]:
```suggestion
				Description:  "The ID of the HCP project where the HCP Consul cluster is located.",
```

File: internal/provider/data_source_consul_agent_kubernetes_secret.go [hunk 2]
```diff
@@ -44,6 +45,14 @@ func dataSourceConsulAgentKubernetesSecret() *schema.Resource {
 				Required:         true,
 				ValidateDiagFunc: validateSlugID,
 			},
+			// Optional inputs
+			"project_id": {
+				Description:  "The ID of the HCP project where the HCP Consul Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/data_source_consul_agent_kubernetes_secret.go:50]:
```suggestion
				Description:  "The ID of the HCP project where the HCP Consul cluster is located.",
```

File: internal/provider/resource_consul_cluster.go [hunk 3]
```diff
@@ -141,17 +142,20 @@ func resourceConsulCluster() *schema.Resource {
 				ForceNew:    true,
 				Computed:    true,
 			},
+			"project_id": {
+				Description:  "The ID of the HCP project where the HCP Consul Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/resource_consul_cluster.go:146]:
```suggestion
				Description:  "The ID of the HCP project where the HCP Consul cluster is located.",
```

File: internal/provider/data_source_consul_cluster.go [hunk 4]
```diff
@@ -28,12 +29,15 @@ func dataSourceConsulCluster() *schema.Resource {
 				Required:         true,
 				ValidateDiagFunc: validateSlugID,
 			},
-			// computed outputs
+			// Optional fields
 			"project_id": {
-				Description: "The ID of the project this HCP Consul cluster is located.",
-				Type:        schema.TypeString,
-				Computed:    true,
+				Description:  "The ID of the HCP project where the HCP Consul Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/data_source_consul_cluster.go:34]:
```suggestion
				Description:  "The ID of the HCP project where the HCP Consul cluster is located.",
```

File: internal/provider/resource_consul_cluster_root_token.go [hunk 5]
```diff
@@ -50,6 +51,15 @@ func resourceConsulClusterRootToken() *schema.Resource {
 				ForceNew:         true,
 				ValidateDiagFunc: validateSlugIDOrID,
 			},
+			// Optional inputs
+			"project_id": {
+				Description:  "The ID of the HCP project where the HCP Consul Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/resource_consul_cluster_root_token.go:56]:
```suggestion
				Description:  "The ID of the HCP project where the HCP Consul cluster is located.",
```

File: internal/provider/resource_consul_snapshot.go [hunk 6]
```diff
@@ -57,12 +58,16 @@ func resourceConsulSnapshot() *schema.Resource {
 				Required:         true,
 				ValidateDiagFunc: validateStringNotEmpty,
 			},
-			// computed outputs
+			// Optional inputs
 			"project_id": {
-				Description: "The ID of the project the HCP Consul cluster is located.",
-				Type:        schema.TypeString,
-				Computed:    true,
+				Description:  "The ID of the HCP project where the HCP Consul Cluster is located.",
```
Comment (bcmdarroch) [inline internal/provider/resource_consul_snapshot.go:63]:
```suggestion
				Description:  "The ID of the HCP project where the HCP Consul cluster is located.",
```

**Review by bcmdarroch — APPROVED** [review]
Just noticed a doc style guide nit

**Review by crhino — APPROVED** [review]
LGTM!

**Pull request merged**